### PR TITLE
ext-ds stubs improvements

### PR DESCRIPTION
**Pull request by enumag**
cc @simPod

### CODE REVIEW

File: stubs/ext-ds.stub [hunk 1]
```diff
@@ -723,3 +714,149 @@ final class Set implements ArrayAccess, Collection, Traversable
...
+	 * @param iterable<TValue>|null $values
+	 */
+	public function __construct($values = null)
+	{
+	}
+
+	/**
+	 * @return Stack<TValue>
+	 */
+	public function copy(): Stack
```
Comment (enumag) [inline stubs/ext-ds.stub]: I'm wondering, should I change cases like these to `self`?

File: stubs/ext-ds.stub [hunk 2]
```diff
@@ -2,51 +2,48 @@
...
 	/**
-	 * @return array<TValue>
+	 * @return array<TKey, TValue>
 	 */
 	public function toArray(): array;
 }
 
 /**
  * @template TValue
  * @implements Sequence<TValue>
```
Comment (simPod) [inline stubs/ext-ds.stub]: Shouldn't we change this to

```suggestion
  * @implements Sequence <int, TValue>
```

as changed here https://github.com/phpstan/phpstan-src/pull/126/files#diff-d0f02c2f44e13de9e215efe917400723R377

File: stubs/ext-ds.stub [hunk 3]
```diff
@@ -517,22 +512,20 @@ interface Sequence extends Collection
...
```
Comment (simPod) [inline stubs/ext-ds.stub]:
```suggestion
  * @implements Sequence <int, TValue>
```

Comment (enumag) [reply stubs/ext-ds.stub]: No, Sequence doesn't have keys.

File: stubs/ext-ds.stub (hunk 2 above)
Comment (enumag) [reply stubs/ext-ds.stub]: I considered it but I don't think so. Sequence can't have keys other than int.

**Comment by ondrejmirtes** [comment]
Thank you!

**Pull request merged**